### DISCUSSION TITLE
Full page component from package fails on hydrate

### DISCUSSION THREAD

**Discussion by jacksleight** [livewire/livewire#9071] [7. Bugs (archived)] [answered]
### Livewire version

v3.5.16

### Laravel version

v11.34.2

### Which PHP version are you using?

PHP 8.3

### What is the problem?

I'm mounting a full page component from a separate package:

```php
<?php

use Illuminate\Support\Facades\Route;
use MyVendor\MyPackage\Livewire\Tools;

Route::get('tools', Tools::class);
```

It loads fine on mount, but as soon as an action runs and it's re-hydrated an error is thrown:

```
Unable to find component: [my-vendor.my-package.livewire.tools]
```

**Edit:**
Registering a name for the component seems to fix it, perhaps this is required? If so it seems odd that it would work on mount.

```php
Livewire::component('my-package-tools', Tools::class);
```

### Code snippets

```php
<?php

namespace MyVendor\MyPackage\Livewire;

use Livewire\Component;

class Tools extends Component
{
    #[Url]
    public $handle;

    public function select($handle)
    {
        $this->handle = $handle;
    }

    public function render()
    {
        return view('my-package::tools');
    }
}
```
```blade
<div>
    <button
        wire:click="select('test')">
        Click
    </button>
</div>
```

### Wirebox URL

not possible in wirebox

### How do you expect it to work?

The component should be re-hydrated successfully.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@jacksleight yeah you need to register your package components because Livewire can't auto-discover them on subsequent requests. The mount works fine as your are referencing the class directly in your route.

**Reply by jacksleight** [reply]
Gotcha, makes sense. Thanks!

**Reply by joshhanley** [reply]
No worries 🙂
